### PR TITLE
sasview: 4.2.0 -> 5.0.4

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/148779

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/misc/sasview/default.nix [hunk 1]
```diff
@@ -1,70 +1,59 @@
...
+  };
 
-  checkPhase = ''
-    # fix the following error:
-    # imported module 'sas.sascalc.data_util.uncertainty' has this __file__ attribute:
-    #   /build/source/build/lib.linux-x86_64-2.7/sas/sascalc/data_util/uncertainty.py
-    # which is not the same as the test file we want to collect:
-    #   /build/source/dist/tmpbuild/sasview/sas/sascalc/data_util/uncertainty.py
-    rm -r dist/tmpbuild
+  doCheck = false; # TODO remove
```
Comment (risicle) [inline pkgs/applications/science/misc/sasview/default.nix:19]: Was this meant to be left in?

File: pkgs/applications/science/misc/sasview/default.nix [hunk 2]
```diff
@@ -1,70 +1,59 @@
...
+  makeWrapperArgs = [
+    "\${qtWrapperArgs[@]}"
+  ];
+
+  checkInputs = with python3.pkgs; [
+    pytestCheckHook
+    unittest-xml-reporting
+  ];
 
-  patches = [ ./pyparsing-fix.patch ./local_config.patch ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/misc/sasview/default.nix:67]: Please also delete the patches.

File: pkgs/applications/science/misc/sasview/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/applications/science/misc/sasview/default.nix:19]: No. I knew I would forget and that's why I added the comment :D

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64.

runs, macos 10.15 (but is hella ugly)

**Pull request merged**